fury/tests/test_primitive.py

```diff
@@ -168,3 +168,48 @@ def test_repeat_primitive_function():
     # big_verts, big_faces, big_colors, big_centers = res
 
     # npt.assert_equal(big_verts.shape[0],  verts.shape[0] * centers.shape[0])
+
+
+def test_build_parametric(npoints=10):
+
+    def surf_equation(u, v):
+        sin = np.sin
+        cos = np.cos
+        x = (1 + v/2 * cos(u/2)) * cos(u)
+        y = (1 + v/2 * cos(u/2)) * sin(u)
+        z = v/2 * sin(u/2)
+        return x, y, z
+    vertices, triangles = fp.build_parametric(0, 1, 0, 1, npoints,
+                                              surf_equation)
+
+    # for npoints>1, following equations follow -
+    # shape of vertices is of the form: (npoints**2, 3)
+    # shape of triangles is of the form: (2*(npoints**2) - 4*(npoints**2) + 2,
+    #                                     3)
+    # for npoints = 10, shape of vertices: (100, 3)
+    # for npoints = 10, shape of triangles: (162, 3)
```

Comment:
what happens if `npoints` is negatif?  In your function above, I recommend you to check this and raise an error. You will need to add a test also.

Reply:
Raised an error and added test for same.